sklearn/metrics/pairwise.py

```diff
@@ -2150,7 +2152,6 @@ def pairwise_distances(
 PAIRWISE_BOOLEAN_FUNCTIONS = [
     "dice",
     "jaccard",
-    "matching",
```

Comment:
Keep this one removed because in Scipy `matching` is a synonym for `hamming`, and `hamming` works for non-boolean values as well.